### ISSUE TITLE
replace eval

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#620]
Where possible, we should replace our calls to eval with dojo.getObject, for dotted object references, or JSON.parse, where JSON is used (would have to be proper JSON with quoted attr names, etc.)  library.js is one example.  The only exception to this I'm aware of is our plugins, which eval actual code, and there was one other piece of code somewhere that used stringified functions for convenience, but probably could be switched over to use code directly.

eval is going to be a security flag, whether or not it's a security issue, so it's best to avoid.  Also, the other methods are more narrow in scope and much more efficient.

**Comment by JonFerraiolo** [comment]
I agree entirely. Regarding our plugins, of course we want to push that into AMD/RequireJS, and let that mechanism invoke eval() or whatever

**Comment by peller** [comment]
Plugins are now AMD. All that's left now for evals are some action references in Workbench.js and some Preference snippet which we may be able to remove.  @aerwin - in your travels, did you happen to learn anything about the eval at Preferences.js:144?

**Comment by peller** [comment]
We should also replace `dojo.fromJSON` with `JSON.stringify` and "proper" JSON data as appropriate -- `dojo.fromJSON` is really an eval and tolerates any JS expression.

**Comment by peller** [comment]
There are also two uses of eval in Context.js to load code in the iframe that could have significant performance impact if we switched over to using the AMD loader.

**Comment by peller** [comment]
@waynevicknair: Note for M5: 4babff0 presents a migration issue.  Existing user library metadata must be converted in this way.

**Comment by peller** [comment]
see also TreeCreateTool

**Comment by peller** [comment]
@waynevicknair has taken care of migration task.  Moving remainder to M6

**Comment by peller** [comment]
We're still using eval to parse djConfig data in Context.js and Theme.js (now require.eval... not sure if that's really any better)

Also, eval is still used in davinci.review.drawing.tools.ExchangeTool

-> M7

**Comment by peller** [comment]
hit all the obvious ones for now. We seem to be stuck with some uses of eval()

-> future
